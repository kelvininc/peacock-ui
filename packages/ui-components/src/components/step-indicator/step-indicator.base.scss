@import '../../assets/styles/globals';

:host {
	/**
	* @prop --indicator-radius: The size of the circle drawn
	* @prop --indicator-default-color: The color to show by default on the indicator
	* @prop --indicator-default-border-color: The border color to show by default on the indicator
	* @prop --indicator-active-color: The color to show when the indicator is active
	* @prop --indicator-active-border-color: The border color to show when the indicator is active
	* @prop --indicator-error-border-color: The color to show on the border that highlights the indicator when there's an error
	*/
	--indicator-radius: 12px;
}

.indicator {
	box-sizing: border-box;
	width: var(--indicator-radius);
	height: var(--indicator-radius);
	border-radius: 50%;
	background-color: var(--indicator-default-color);
	border: 1px solid var(--indicator-default-border-color);
	transition: all 0.25s ease;

	&--state-enabled {
		cursor: pointer;
	}

	&--state-active {
		background-color: var(--indicator-active-color);
		border-color: var(--indicator-active-border-color);
	}

	&--state-error {
		border-color: var(--indicator-error-border-color);
	}
}
