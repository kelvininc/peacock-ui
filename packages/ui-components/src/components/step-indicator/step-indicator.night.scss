@import './step-indicator.base';

:host {
	--indicator-default-color: #{kv-color('neutral-5')};
	--indicator-default-border-color: #{kv-color('neutral-6')};
	--indicator-active-color: #{kv-color('neutral-2')};
	--indicator-active-border-color: #{kv-color('primary')};
	--indicator-error-border-color: #{kv-color('error')};
}
