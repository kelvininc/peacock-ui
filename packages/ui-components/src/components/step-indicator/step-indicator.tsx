import { Component, Host, Prop, Event, EventEmitter, h } from '@stencil/core';
import { IStepIndicator, IStepIndicatorEvents } from './step-indicator.types';

@Component({
	tag: 'kv-step-indicator',
	styleUrls: {
		night: 'step-indicator.night.scss'
	},
	shadow: true
})
export class KvStepIndicator implements IStepIndicator, IStepIndicatorEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) stepKey?: number;
	/** @inheritdoc */
	@Prop({ reflect: true }) enabled?: boolean;
	/** @inheritdoc */
	@Prop({ reflect: true }) active?: boolean;
	/** @inheritdoc */
	@Prop({ reflect: true }) hasError?: boolean;

	/** @inheritdoc */
	@Event() indicatorClicked: EventEmitter<number>;

	private onIndicatorClick = () => {
		if (this.enabled) {
			this.indicatorClicked.emit(this.stepKey);
		}
	};

	render() {
		return (
			<Host>
				<div
					class={{
						'indicator': true,
						'indicator--state-enabled': this.enabled,
						'indicator--state-active': this.active,
						'indicator--state-error': this.hasError
					}}
					onClick={this.onIndicatorClick}
				></div>
			</Host>
		);
	}
}
