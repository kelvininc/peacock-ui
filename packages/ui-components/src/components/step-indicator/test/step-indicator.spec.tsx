import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvStepIndicator } from '../step-indicator';

describe('StepIndicator (unit tests)', () => {
	let page: SpecPage;
	let component: KvStepIndicator;

	describe('when rendering with default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStepIndicator],
				html: `<kv-step-indicator></kv-step-indicator>`
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});
		});
	});

	describe('when rendering with `active`', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStepIndicator],
				html: `<kv-step-indicator active></kv-step-indicator>`
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});

			it('should set the `active` prop to true', () => {
				expect(component.active).toEqual(true);
			});
		});
	});

	describe('when rendering with `hasError`', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStepIndicator],
				html: `<kv-step-indicator has-error></kv-step-indicator>`
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});

			it('should set the `active` prop to true', () => {
				expect(component.hasError).toEqual(true);
			});
		});
	});
});
