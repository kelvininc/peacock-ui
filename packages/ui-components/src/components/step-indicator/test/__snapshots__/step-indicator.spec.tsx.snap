// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator (unit tests) when rendering with \`active\` when the component loads should match the snapshot 1`] = `
<kv-step-indicator active="">
  <mock:shadow-root>
    <div class="indicator indicator--state-active"></div>
  </mock:shadow-root>
</kv-step-indicator>
`;

exports[`StepIndicator (unit tests) when rendering with \`hasError\` when the component loads should match the snapshot 1`] = `
<kv-step-indicator has-error="">
  <mock:shadow-root>
    <div class="indicator indicator--state-error"></div>
  </mock:shadow-root>
</kv-step-indicator>
`;

exports[`StepIndicator (unit tests) when rendering with default props when the component loads should match the snapshot 1`] = `
<kv-step-indicator>
  <mock:shadow-root>
    <div class="indicator"></div>
  </mock:shadow-root>
</kv-step-indicator>
`;
